File: modulo-govsocial/api/app/schemas/agenda.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class AppointmentCreate(BaseModel):
    unit_id: uuid.UUID
    professional_id: Optional[uuid.UUID] = None
    person_id: Optional[uuid.UUID] = None
    family_id: Optional[uuid.UUID] = None
    tipo: str = "ATENDIMENTO"
    data_hora_inicio: datetime
    data_hora_fim: Optional[datetime] = None
    observacoes: Optional[str] = None
    opt_in_lembrete: bool = False


class AppointmentUpdate(BaseModel):
    professional_id: Optional[uuid.UUID] = None
    status: Optional[str] = None
    data_hora_inicio: Optional[datetime] = None
    data_hora_fim: Optional[datetime] = None
    observacoes: Optional[str] = None


class AppointmentOut(BaseModel):
    id: uuid.UUID
    unit_id: uuid.UUID
    professional_id: Optional[uuid.UUID]
    person_id: Optional[uuid.UUID]
    family_id: Optional[uuid.UUID]
    tipo: str
    status: str
    data_hora_inicio: datetime
    data_hora_fim: Optional[datetime]
    observacoes: Optional[str]
    senha: Optional[str]
    lembrete_enviado: bool
    opt_in_lembrete: bool
    created_at: datetime
    updated_at: datetime


class SenhaChamada(BaseModel):
    """Chamar próximo da fila."""
    professional_id: uuid.UUID


class VisitaCreate(BaseModel):
    family_id: uuid.UUID
    unit_id: uuid.UUID
    professional_id: Optional[uuid.UUID] = None
    data_planejada: datetime
    observacoes: Optional[str] = None


class VisitaUpdate(BaseModel):
    status: Optional[str] = None
    data_realizada: Optional[datetime] = None
    endereco_confirmado: Optional[str] = None
    observacoes: Optional[str] = None
    attendance_id: Optional[uuid.UUID] = None


class VisitaOut(BaseModel):
    id: uuid.UUID
    family_id: uuid.UUID
    unit_id: uuid.UUID
    professional_id: Optional[uuid.UUID]
    attendance_id: Optional[uuid.UUID]
    data_planejada: datetime
    data_realizada: Optional[datetime]
    status: str
    endereco_confirmado: Optional[str]
    observacoes: Optional[str]
    created_at: datetime
    updated_at: datetime

File: modulo-govsocial/web/src/tipos/agenda.ts
/**
 * DTOs de agenda, fila do dia e visitas domiciliares (Fase 7).
 * Espelham app/schemas/agenda.py e app/schemas/prontuario.py (ReceptionOut).
 */

export type AppointmentOut = {
  id: string;
  unit_id: string;
  professional_id: string | null;
  person_id: string | null;
  family_id: string | null;
  tipo: string; // ATENDIMENTO | VISITA_DOMICILIAR | GRUPO
  status: string; // AGENDADO | AGUARDANDO | EM_ATENDIMENTO | CONCLUIDO | FALTOU | CANCELADO
  data_hora_inicio: string;
  data_hora_fim: string | null;
  observacoes: string | null;
  senha: string | null;
  lembrete_enviado: boolean;
  opt_in_lembrete: boolean;
  created_at: string;
  updated_at: string;
};

export type AppointmentCreate = {
  unit_id: string;
  professional_id?: string | null;
  person_id?: string | null;
  family_id?: string | null;
  tipo?: string;
  data_hora_inicio: string;
  data_hora_fim?: string | null;
  observacoes?: string | null;
  opt_in_lembrete?: boolean;
};

export type AppointmentUpdate = {
  professional_id?: string | null;
  status?: string;
  data_hora_inicio?: string;
  data_hora_fim?: string | null;
  observacoes?: string | null;
};

export type VisitaOut = {
  id: string;
  family_id: string;
  unit_id: string;
  professional_id: string | null;
  attendance_id: string | null;
  data_planejada: string;
  data_realizada: string | null;
  status: string;
  endereco_confirmado: string | null;
  observacoes: string | null;
  created_at: string;
  updated_at: string;
};
